junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java

```diff
@@ -153,6 +158,30 @@ private RunListener createRunListener(String runListenerClassName) {
     } else return null;
   }
 
+  /**
+   * Alter default RunSettings depending on the passed selectors. If selectors contains only
+   * elements of type TestSelector, then default settings are altered to include only test names
+   * from these selectors. This allows to run particular test cases within given test class.
+   * testFilter is treated as a regular expression, hence joining is done via '|'.
+   */
+  private RunSettings alterRunSettings(RunSettings defaultSettings, Selector[] selectors) {
+    boolean onlyTestSelectors =
+        Arrays.stream(selectors).allMatch(selector -> selector instanceof TestSelector);
+    if (onlyTestSelectors) {
+      String testFilter =
+          Arrays.stream(selectors)
+              .map(selector -> ((TestSelector) selector).testName())
```

Comment:
Should we guard this cast?

Reply:
I think I can try to get rid off this cast